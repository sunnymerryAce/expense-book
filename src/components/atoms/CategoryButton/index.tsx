import React from "react";
import Chip from "@material-ui/core/Chip";

export interface CategoryButtonProps {
  selected: boolean;
  label: string;
  onClick: (props: any) => any;
}

const CategoryButton: React.FC<CategoryButtonProps> = (props) => {
  return (
    <Chip
      color={props.selected ? "secondary" : "primary"}
      label={props.label}
      onClick={props.onClick}
    />
  );
};

export default CategoryButton;
