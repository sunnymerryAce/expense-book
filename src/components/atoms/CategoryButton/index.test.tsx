import React from "react";
import CategoryButton from "./";
import { shallow } from "enzyme";

describe("<CategoryButton />", () => {
  let container = null;

  const props = {
    selected: false,
    label: "Food",
    handleClick: jest.fn(),
  };

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    container = shallow(<CategoryButton {...props} />);
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    container = shallow(<CategoryButton {...props} />);
    expect(container.find(".CategoryButton").props()).toMatchObject({
      color: "primary",
      label: "Food",
      onClick: props.handleClick,
    });
  });

  test("should have primary colors", () => {
    container = shallow(<CategoryButton {...props} />);
    expect(container.find(".CategoryButton").props()).toMatchObject({
      color: "primary",
    });
  });

  test("should have secondary colors", () => {
    const props = {
      selected: true,
      label: "Food",
      handleClick: jest.fn(),
    };
    container = shallow(<CategoryButton {...props} />);
    expect(container.find(".CategoryButton").props()).toMatchObject({
      color: "secondary",
    });
  });

  test("click event handler should be triggered", () => {
    container = shallow(<CategoryButton {...props} />);
    container.simulate("click");
    expect(props.handleClick).toBeCalled();
  });
});
