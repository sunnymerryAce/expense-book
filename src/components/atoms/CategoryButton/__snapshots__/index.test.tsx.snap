// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryButton /> should match the snapshot 1`] = `"<div class=\\"MuiButtonBase-root MuiChip-root CategoryButton MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable\\" tabindex=\\"0\\" role=\\"button\\"><span class=\\"MuiChip-label\\">Food</span></div>"`;
